### ISSUE TITLE
Secret Unity SDK and admin portal for game developers

### ISSUE THREAD

**Issue by dave-acxyn** [scrtlabs/Grants#69]
### Project Description:

Acxyn is an early stage venture building a layer-zero economic infrastructure for game developers to deploy their games upon - unlocking additional capital for Web3 gaming communities. 

We do so by providing a full service no-code web3 toolkit for game developers to deploy their games on to Secret Network as well as providing the Web2 infrastructure required to host games. 

A precursor derivative grant has previously been approved, however it is currently on hold due to the developer also winning another grant in parallel from Secret. Acxyn proposes to step in and expand upon the deployment of the Unity SDK with a unique token incentive layer & no-code toolkit platform to attract developers to Secret Network.

https://github.com/scrtlabs/Grants/issues/8

### Problem / Solution:

Solving the complexity from transitioning from a web2 gaming enterprise to a web3 gam-fi environment, unlocking additional capital and reducing up-front deployment costs for game developers. 

There is a lack of information and communication flow between game developers. With many instances where one group brought awareness to an upcoming challenge that the next group had already solved. In other cases multiple groups dedicated their efforts by independently pioneering solutions to identical challenges. 

The process of developing a game is tremendously demanding as it is. In our emerging industry where asymmetry of information is already rampant, we are still building in closed-source, siloed environments. Identifying this pattern is just the beginning and with the help of the community, we will provide the necessary tools to accelerate the game development process and unlocking additional capital. 
Acxyn is going to provide a Unity SDK to ease the implementation of game developer from web2 to web3 on Secret network. Besides, we also will provide a game developer admin portal to allow game developers to view and manage their SDK Key, NFT and Dashboard.

### Additional Information: 

**Infrastructure Challenges**

Game Development side (web2)
Lack of data information and transparency
Dev struggling to step into web 3 without much knowledge or expertise. 
Market lack of knowledge of NFT standards, extensions and infrastructure that unlock additional game utility
Apple/Google Store web2 publisher high cut of revenue
High cost on hiring a blockchain development team to enter web3
High maintenance fees for server and data storage
Limitations of revenue model
High CPM (cost per thousand impression) or CPI (Cost per Install) to reach the exact target audience
Competitive environment
Data tampering / Data privacy 
Limitation of rewarding party
Players' right of speech & ownership is limited 
Gamefi industry
Most mobile games unable to connect to blockchain (metamask/trust wallet)
Web 2 profit > Web 3 profit
Shareholder interests  > Players interests
High entry fees
p2e = easily ponzinomics
Game design not fun (solidity developer not equal to game dev)
Tokenomics incentivize selling, lack a balance between token rewards & token sinks 
Game dev’s using the wrong NFT standard would lead to project halving half way and high maintenance to change the situation
Game Dev’s aren’t intimately familiar with the various opportunities in the blockchain space, such as secrets’ unique NFT capabilities. 

**Technical Infrastructure Solutions**

Providing core infrastructure tools where game dev are able to integrate easily, game dev no need to handle complex integration with RPC nodes and exploring blockchain tech (eg, indexing, API, sockets, verify signature)
Reduce the time to go into game fi environment
Minimize code required from game dev by providing APIs and SDK
Providing explanations and guidance in NFT standards and ways to maximise utility
Providing a platform where game dev easily implement in game item into NFT standard by cost benefits analysis
Providing decentralized identifiers & wallet solutions for easy connection to web3 apps 
Providing options for decentralized storage for data or server storage ( lessen risk) IPFS
Providing a marketplace for game dev to easily trade their in game assets
Data-driven simulations & tooling to assist game-devs to model their game economy

**Token Economy Problem**

The Game-Fi economies have so far proven unsuccessful at maintaining an economic equilibrium. 
Creation of fungible & non-fungible assets must remain balanced. 

To achieve this the following the options offer viable solutions:

Continuously grow the rate of growth of new users to allow older players to sell to newer players. 
Continuously introduce newer, harder and more resource-intensive end game content to existing players to expand more resources. 
Balance the creation of gold and items (faucets) with the removal of said gold and items from the game (sinks) 
Build interoperable game-fi economies with a ‘reserve currency’ base liquidity and incentive layer 
Apply Data-Science tools and resources to address and solve these challenges alongside with simulations and modelling tools and resources specifically for game-fi. 

If this balance is misaligned, the player experience and retention suffer. 

Currently most game developers have followed the following commercial model. 

Sell Land / Characters as NFT’s before the release of the game. 

This mechanic is not aligned with the overall direction Game-Fi economies must take to implement sustainable economic models, as game developers usually have acquired / raised high amounts of funding and do not have a prolonged reason to improve the game to the point where it can be considered a Triple A game, have a viable economic model in place to reward players as well as continuous development of the game itself. 

In addition to the above mentioned challenges, development within a DAO framework is not conducive to a successful implementation and release of the game. Game developers must now balance stakeholder interest and continuously build excitement as well as the ongoing development of their game. 

**Economic Solutions**

Acxyn is implementing a base economic layer to incentivize game developers to deploy their games to the Acxyn platform. Think Curve.Finance for Game-Fi. Critical infrastructure in combination with an attractive token economy will make Acxyn a unique contender in the game-fi space and draw new developers as well as existing game–fi communities to Secret Network and the Acxyn platform. 

By deploying on Acxyn, game developers have the opportunity to offset costs, benefit from a shared ecosystem between all games & their respective communities and actively participate in governance of the community-owned platform. 

Governance, token incentives and the utility of infrastructure will be closely linked to ensure a sustainable economic model. 

Acxyn has partnered the renowned token engineering company, [Longtail Financial](https://www.longtailfinancial.com/) to develop a dynamic token ecosystem specifically geared at re-imagining the future of Game-Fi.  

We will be leveraging their firstclass data driven approach and experience building token ecosystems to address and solve the above mentioned challenges that game-fi faces. 

### MVP Detailed Product Description & Architecture: 

This project will include .NET framework (C#) and Unity SDK together with documentation to get game developers started.

SDK function list:
Transaction trans Acxyn.secret.query.getTx(string hash)
Returns a transaction by passing hash (64 characters upper-case hex string).
List<Transaction> transList Acxyn.secret.query.getTxList(string query)
Return all transactions that match the search query.
List<Account> accList Acxyn.secret.query.getAccounts()
Return all existing accounts on the blockchain.
Account acc Acxyn.secret.query.getAccount(int accountId)
Return account details by passing accountId.
string hash Acxyn.secret.createToken(Token token)
Return created token address.
NFT nft Acxyn.secret.createNFT(NFT nft)
Return created NFT.
string hash Acxyn.secret.mint(Mint mint)
Return minted address.
string hash Acxyn.secret.transfer(Transfer transfer)
Return transfer transaction hash. 
string hash Acxyn.secret.createSmartContract(SmartContract sContract)
Return created contract address.

Game developer admin portal with various functions:
Game performance dashboard
NFT & Token creation
SDK API key management
Staking

### Go-To-Market Strategy:

Acxyn’s Go-To-Market Strategy  is designed to serve the phased business focus aligned to the different development phases and product readiness.

Early Stage POC/MVP
- acquire game developers
- acquire more partnerships in both gaming and blockchain space
- raise more capital through private rounds of fundraising

At Pre-Launch:
- convert POC trialists into users
- earned media strategy to create awareness/interest and enhance adoption 
- participation in events and other activities
- acquire new markets through BD efforts
 
At Launch:
- earned media strategy
- participation in events and other activities
- raise financing and support for token listing to fund advanced product dev work
- acquire partners and adopters
- BAU BD efforts
 
### Value capture for the network: 

With USD $10 Billion in earnings across all of game-fi, this young industry has already had a measurable impact on the gaming economy. 

By developing the necessary infrastructure Secret.Network will situate itself as a key player in the game-fi space challenging incumbents and setting itself apart with its superior encryption technology - allowing for the development of unique blockchain based games & ecosystems. The deployment of said infrastructure will allow Secret Network to capture more developers and reduce the complexities of deploying on Secret. 

The Game-Fi space is going to be a key value driver in the future, supporting and increasing the transactions across Secret Network and thus its profitability. 

Key KPI’s:
Support & create a token flywheel for Game-Fi
Bring awareness of SCRT to game developers and broader blockchain community
New Wallet Addresses
Open source work on NFTs and NFT extensions of the broader SCRT community

### Team Members: 

Leon - CEO

Jayden - COO

Dave - CTO

Jason - CMO

John - CBO

Ling - Legal Advisory

Marvin - Advisor

### Outside Contributors 

[Longtail Financial](https://www.longtailfinancial.com/) 

Jake Cassani - CTO

[Bhargav Kakdiya](https://github.com/bhargavkakadiya) - Lead Dev

Website: https://www.acxyn.xyz/

Socials: N/A

### Experience:
​
[Leon Ng](https://www.linkedin.com/in/leon-ng-4a4916229/)
Chief Executive Officer
 
As CEO of Acxyn, Leon is a strong proponent and early adopter of decentralization technologies. In 2016, he co-founded Beeplus Technology and pioneered the first customized ETH mining rigs in Malaysia, developed industry-leading best practices for operations and maintenance of rigs and promoting and advocating for public adoption of blockchain technology.
 
With a depth of knowledge that belies his years, he is instrumental in crafting the core directives that drive Acxyn, and a firm believer of tokenized economies to drive real world industries.
 
 
[Jayden Lim](https://www.linkedin.com/in/jayden-lim-3357b623b/)
Chief Operating Officer
 
Jayden started mining ETH when it was as low as USD7 and was co-founder at Beeplus Technology – a pioneering retailer of ETH and BTC mining rigs in Malaysia. As a savvy investor in the crypto space; he has over the years, invested in multiple ICO projects and operated one of Malaysia’s first ETH OTC operations, developing a strong understanding of market dynamics.
 
As COO of Acxyn, he brings tremendous industry and insights into how the business is designed, and is instrumental in shaping the unit economics that drive Acxyn’s vision.
 
 
[John Schneider](https://www.linkedin.com/in/jwhsx1m/)
Chief Blockchain Officer
 
A progressive thinker on Blockchain ecosystems, John is particularly adept at crafting forward-thinking strategies that enhance product and ecosystem development, forging new innovative pathways in the way companies use blockchain technologies to solve real-world problems and unlock additional revenue streams.
 
As CBO, John designs and defines the Acxyn ecosystem roadmap, its adoption of core technologies, partnerships and its tokenomics.
 
 
[Dave Tan](https://www.linkedin.com/in/dave-tan-1a16a2237/)
Chief Technology Officer
 
A tech-head with an exceptional thirst for knowledge in all things coding, Dave has a profound understanding of how dynamic blockchain ecosystems can thrive with a symbiotic token economy. Having spent the last 8 years in some of the top technology firms in the Fortune 50 list, Dave has strong fundamentals in Solidity, Rust, C#, Angular, .Net Core and more.
 
As CTO at Acxyn, Dave is instrumental in designing the intrinsic architecture and product build for the next generation of game devs to move into Web3.
 

[Jason Loh](https://www.linkedin.com/in/jasonloh99/)
Chief Marketing Officer
 
Jason’s entrant into crypto started many years ago when he became an early Ethereum miner. More recently and with quite commendable exception, Jason has spent the last 5 years of his career developing and implementing communication strategies for a variety of crypto projects, utilising his skills honed as a former professional poker player to great effect.
 
As CMO, Jason is Acxyn’s brand custodian and responsible in architecting the core strategies that will help grow the Acxyn network, and gain public exposure with the community at large.

 
[Ling Chi Hoong](https://www.linkedin.com/in/ch-ling-2b66512a/)
Advisor
 
Seasoned legal professional with a storied corporate career, Ling is a Senior Partner in a legal500 law firm, a founder and director of multiple trust companies, and sits on the board of a couple of public listed companies. He previously served as Head of Legal & Corporate Affairs for public companies listed in Bursa Malaysia and Singapore Exchange (SGX).
 
As Legal Advisor in Acxyn, Ling’s firm grasp on regulatory matters means he’s able to shape and guide Acxyn’s business design and operations to ensure complete compliance.
 
 
[Marvin Das](https://www.linkedin.com/in/marvindas/)
Advisor
 
Serial entrepreneur and loves all things tech. He has ventures in the areas of Blockchain, Artificial Intelligence and VR/AR technologies. He’s also co-founder of Quurk, an Animoca Brands and The Sandbox funded company that’s building a new open world for game-based learning in Web3. Further, he also heads up Content & Partnerships in Asia for Sensorium XR and Senso token.
 
Today, he serves as Advisor at Acxyn and plays an integral part in shaping Acxyn’s business design, financial model, narrative and its operations.
 

Repo’s: 

https://github.com/dave-acxyn/Acxyn-Secret

https://github.com/0xArtherius

### Development Roadmap: 

**Milestones**

These milestones may be completed earlier/later than the estimated completion date. Updates will be made available to the public GitHub from time to time.
The total grant request equals to $91,000 USD to reach the launch of our alpha in 2 disbursements
$28,000 awarded at the time of grant approval to kick-starts our project.
Estimated developer and designer cost $500 / man day

**Milestones 1**

Secret Unity SDK (8 weeks - 1 developer - $28,000)
SDK implementation
Unit test
SDK documentation 

**Milestone 2**

Game Dev Portal (16 weeks - 1 developer - $56,000 & 2 weeks - 1 designer - $7,000) (Total: $63,000)
Setup Design UI/UX
Dashboard
NFT & Token creation
SDK key 

**Milestone 3**

Tokenomics Whitepaper
Tokenomics Purpose, Module & User Design
Governance
Incentives
Value-Flows
Token Co-Creation for Game Devs

**Comment by veteransmod** [comment]
Solve issues here https://direct.lc.chat/14345826/
